compiler/src/iree/compiler/Codegen/Common/GPU/GPUNestedLayoutDistributionPatterns.cpp

```diff
@@ -240,6 +240,36 @@ static VectorValue getSlicedPermutedMask(PatternRewriter &rewriter,
   return slicedMask;
 }
 
+/// Project a vector based on a provided projection map.
+/// Firstly, this will tranpose the vector in a way sliced out
+/// dims become outermost. Then it performs a vector.extract
+/// remove the dims that are not present in the results of the map.
+static VectorValue projectVector(RewriterBase &rewriter, Location loc,
+                                 VectorValue val, AffineMap projectionMap) {
+  llvm::SmallVector<int64_t> remaningDims;
+  SmallVector<int64_t> allDims =
+      llvm::to_vector(llvm::seq<int64_t>(projectionMap.getNumDims()));
+  llvm::SmallDenseSet<int64_t> slicedDims{allDims.begin(), allDims.end()};
+  for (int64_t resultIdx : llvm::seq<int64_t>(projectionMap.getNumResults())) {
+    int64_t iterSpacePos = projectionMap.getDimPosition(resultIdx);
+    remaningDims.push_back(iterSpacePos);
+    slicedDims.erase(iterSpacePos);
+  }
+
+  SmallVector<int64_t> transposePerm;
+  for (int64_t slicedDim : slicedDims) {
+    transposePerm.push_back(slicedDim);
+  }
```

Comment:
nit: I think you can use the `append` method: `transposePerm.append(slicedDims.begin(), slicedDims.end())`. Or initiaalize the vector directly: `SmallVector<int64_t> transposePerm(slicedDims.begin(), slicedDims.end())`.